File: src/main/java/com/kec/hms/service/AppointmentService.java
package com.kec.hms.service;

import com.kec.hms.model.Appointment;
import com.kec.hms.model.Doctor;
import com.kec.hms.model.Patient;
import com.kec.hms.repository.AppointmentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDate;
import java.util.List;

@Service
public class AppointmentService {

    @Autowired
    private AppointmentRepository appointmentRepository;

    /**
     * Get all appointments
     */
    public List<Appointment> getAllAppointments() {
        return appointmentRepository.findAll();
    }

    /**
     * Get appointment by ID
     * FIXED: Changed from int to Long
     */
    public Appointment getAppointmentById(Long id) {
        return appointmentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Appointment not found with id: " + id));
    }

    /**
     * Create new appointment
     */
    public Appointment createAppointment(Appointment appointment) {
        if (appointment.getStatus() == null) {
            appointment.setStatus("SCHEDULED");
        }
        return appointmentRepository.save(appointment);
    }

    /**
     * Update appointment
     * FIXED: Changed from int to Long
     */
    public Appointment updateAppointment(Long id, Appointment appointmentDetails) {
        Appointment appointment = getAppointmentById(id);
        
        appointment.setAppointmentDate(appointmentDetails.getAppointmentDate());
        appointment.setAppointmentTime(appointmentDetails.getAppointmentTime());
        appointment.setReason(appointmentDetails.getReason());
        appointment.setNotes(appointmentDetails.getNotes());
        appointment.setStatus(appointmentDetails.getStatus());
        
        return appointmentRepository.save(appointment);
    }

    /**
     * Delete appointment
     * FIXED: Changed from int to Long
     */
    public void deleteAppointment(Long id) {
        appointmentRepository.deleteById(id);
    }

    /**
     * Get all appointments for a patient
     */
    public List<Appointment> getAppointmentsByPatient(Patient patient) {
        return appointmentRepository.findByPatient(patient);
    }

    /**
     * Get all appointments for a doctor
     */
    public List<Appointment> getAppointmentsByDoctor(Doctor doctor) {
        return appointmentRepository.findByDoctor(doctor);
    }

    /**
     * Get appointments by date
     */
    public List<Appointment> getAppointmentsByDate(LocalDate date) {
        return appointmentRepository.findByAppointmentDate(date);
    }

    /**
     * Get appointments by status
     */
    public List<Appointment> getAppointmentsByStatus(String status) {
        return appointmentRepository.findByStatus(status);
    }

    /**
     * Get doctor's appointments for a specific date
     */
    public List<Appointment> getDoctorAppointmentsByDate(Doctor doctor, LocalDate date) {
        return appointmentRepository.findByDoctorAndAppointmentDate(doctor, date);
    }

    /**
     * Get patient's appointments by status
     */
    public List<Appointment> getPatientAppointmentsByStatus(Patient patient, String status) {
        return appointmentRepository.findByPatientAndStatus(patient, status);
    }

    /**
     * Cancel appointment
     * FIXED: Changed from int to Long
     */
    public Appointment cancelAppointment(Long id) {
        Appointment appointment = getAppointmentById(id);
        appointment.setStatus("CANCELLED");
        return appointmentRepository.save(appointment);
    }

    /**
     * Complete appointment
     * FIXED: Changed from int to Long
     */
    public Appointment completeAppointment(Long id) {
        Appointment appointment = getAppointmentById(id);
        appointment.setStatus("COMPLETED");
        return appointmentRepository.save(appointment);
    }
}

File: HMS-frontend/src/services/appointmentService.js
// src/services/appointmentService.js
import axiosInstance from '../api/axios';

const appointmentService = {
  // ============================================
  // PATIENT ENDPOINTS
  // ============================================
  
  // Get my appointments (as patient)
  getMyAppointments: async () => {
    const response = await axiosInstance.get('/api/v1/appointments/my');
    return response.data;
  },
  
  // Book new appointment (as patient)
  bookAppointment: async (appointmentData) => {
    const response = await axiosInstance.post('/api/v1/appointments/book', appointmentData);
    return response.data;
  },
  
  // Cancel my appointment
  cancelAppointment: async (id) => {
    const response = await axiosInstance.put(`/api/v1/appointments/${id}/cancel`);
    return response.data;
  },
  
  // ============================================
  // DOCTOR ENDPOINTS
  // ============================================
  
  // Get my appointments (as doctor)
  getMyDoctorAppointments: async () => {
    const response = await axiosInstance.get('/api/v1/appointments/doctor/my');
    return response.data;
  },
  
  // Mark appointment as completed
  completeAppointment: async (id) => {
    const response = await axiosInstance.put(`/api/v1/appointments/${id}/complete`);
    return response.data;
  },
  
  // ============================================
  // ADMIN ENDPOINTS
  // ============================================
  
  // Get all appointments (admin)
  getAllAppointments: async () => {
    const response = await axiosInstance.get('/api/v1/appointments');
    return response.data;
  },
  
  // Create appointment (admin/doctor)
  createAppointment: async (appointmentData) => {
    const response = await axiosInstance.post('/api/v1/appointments', appointmentData);
    return response.data;
  },
  
  // Update appointment
  updateAppointment: async (id, appointmentData) => {
    const response = await axiosInstance.put(`/api/v1/appointments/${id}`, appointmentData);
    return response.data;
  },
  
  // Delete appointment
  deleteAppointment: async (id) => {
    const response = await axiosInstance.delete(`/api/v1/appointments/${id}`);
    return response.data;
  },
};

export default appointmentService;